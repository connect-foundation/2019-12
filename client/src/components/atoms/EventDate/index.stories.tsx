import React from 'react';

import EventDate from '.';

export default {
  title: 'Atoms / EventDate',
};

export const samedate: React.FC = () => (
  <EventDate startAt={'2018-04-22T00:00:00Z'} endAt={'2018-04-22T09:00:00Z'} />
);

export const diffdate: React.FC = () => (
  <EventDate startAt={'2019-12-21T03:00:00Z'} endAt={'2019-12-22T09:00:00Z'} />
);
