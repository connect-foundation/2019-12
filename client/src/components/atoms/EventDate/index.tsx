import React from 'react';

import * as S from './style';
import { calculateStringOfDateRange } from 'utils/dateCalculator';
import { getKoreanDateString } from '../../../utils/dateCalculator/index';

interface EventDateProps {
  startAt?: string;
  endAt?: string;
  children?: string;
}

function EventDate({
  startAt,
  endAt,
  children,
}: EventDateProps): React.ReactElement {
  if (startAt && endAt) {
    return <>{calculateStringOfDateRange(startAt, endAt)}</>;
  }
  return <>{children ? getKoreanDateString(children) : ''}</>;
}

export default EventDate;
