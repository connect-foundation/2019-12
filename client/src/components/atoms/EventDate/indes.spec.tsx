import React from 'react';
import { mount } from 'enzyme';
import EventDate from '.';

describe('Atom / Divider', () => {
  it('[SNAPSHOT] 렌더링 (same date)', () => {
    // given
    const wrapper = mount(
      <EventDate
        startAt={'2018-04-22T00:00:00Z'}
        endAt={'2018-04-22T09:00:00Z'}
      />,
    );

    // then
    expect(wrapper).toMatchSnapshot();
  });

  it('[SNAPSHOT] 렌더링 (diff date)', () => {
    // given
    const wrapper = mount(
      <EventDate
        startAt={'2019-12-21T03:00:00Z'}
        endAt={'2019-12-22T09:00:00Z'}
      />,
    );

    // then
    expect(wrapper).toMatchSnapshot();
  });
});
