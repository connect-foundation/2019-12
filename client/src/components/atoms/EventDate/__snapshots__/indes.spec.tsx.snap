// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom / Divider [SNAPSHOT] 렌더링 (diff date) 1`] = `
<EventDate
  endAt="2019-12-22T09:00:00Z"
  startAt="2019-12-21T03:00:00Z"
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "bDWFJH",
            "rules": Array [],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className=""
      >
        2019년 12월 21일 (토) 오후 12:00
- 12월 22일 (일) 오후 06:00
      </div>
    </StyledComponent>
  </styled.div>
</EventDate>
`;

exports[`Atom / Divider [SNAPSHOT] 렌더링 (same date) 1`] = `
<EventDate
  endAt="2018-04-22T09:00:00Z"
  startAt="2018-04-22T00:00:00Z"
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "bDWFJH",
            "rules": Array [],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className=""
      >
        2018년 04월 22일 (일)
오전 09:00 - 오후 06:00
      </div>
    </StyledComponent>
  </styled.div>
</EventDate>
`;
